### PR TITLE
[W5.3b][T15-A3] Dickson Chan

### PR DESCRIPTION
**Pull request by itsdickson**
created a super-class, Contact.java
removed common variables and methods from Address.java, Phone.java, Email.java

### CODE REVIEW

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -6,27 +6,23 @@
...
-    public final String value;
-    private boolean isPrivate;
-
     /**
      * Validates given address.
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
+        super(address, isPrivate);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: here you are trying to assign value before validating it.

File: src/seedu/addressbook/data/person/Contact.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java]: Need header comments for the class and its methods (if the methods are doing some non-trivial stuff)

File: src/seedu/addressbook/data/person/Contact.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    public String value;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java]: try to make it private. in general, start off with private members, then use protected or public only if absolutely necessary.

File: src/seedu/addressbook/data/person/Contact.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof Contact // instanceof handles nulls
+                && this.value.equals(((Contact) other).value)); // state check
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Contact.java]: This can potentially equate a phone number with address, think why?

**Comment by okkhoy** [comment]
@itsdickson 
Some comments added. Please ack & close the PR after reading comments.

File: src/seedu/addressbook/data/person/Contact.java (hunk 3 above)
Comment (itsdickson) [reply src/seedu/addressbook/data/person/Contact.java]: thanks for the reminder! i'll definitely take note of it.

**Pull request closed without merging**